File: csharp/DiagramPrinter/DiagramMetadata.cs
namespace DiagramPrinter;

public class DiagramMetadata
{
    public string FullFilename { get; protected set; }
    public string FileType { get; protected set; }
    public bool FileAvailable { get; protected set; }

    public DiagramMetadata(IDiagram idiagram)
    {
        FullFilename = idiagram.Name() + "_" + idiagram.SerialNumber();
        FileType = idiagram.Name().Contains("Flowchart") ? "PDF" : "Spreadsheet";
        FileAvailable = !idiagram.IsDisposed();
    }

    protected DiagramMetadata()
    {
        // for subclass for testing
    }
}

File: kotlin/src/main/kotlin/sammancoaching/DiagramMetadata.kt
package sammancoaching

open class DiagramMetadata {
    var fullFilename: String
    var fileType: String
    protected var fileAvailable: Boolean

    constructor(diagram: FlowchartDiagram) {
        this.fullFilename = diagram.name + "_" + diagram.serialNumber
        this.fileType = if (diagram.name.contains("Flowchart")) "PDF" else "Spreadsheet"
        this.fileAvailable = !(diagram.isDisposed)
    }

    protected constructor() {
        // Enable subclassing for testing purposes
        this.fullFilename = ""
        this.fileType = ""
        this.fileAvailable = false
    }
}